parse-report.js

```diff
@@ -12,20 +12,26 @@ const durationRegex = /duration_ms\s([\d.]+)/
 export default async function parseReport(source) {
   const tests = []
   const testStack = []
-
-  let diagnosticMessage = ''
+  const allTests = Object.create(null)
+  let currentTest
   let totalDuration = 0
 
-  function lastTestInStack() {
-    return testStack.length ? testStack[testStack.length - 1] : null
+  function testId(data) {
+    return [data.file, data.line, data.column].join('\0')
   }
 
-  function appendDiagnosticMessage(message) {
-    diagnosticMessage += `${message}\n`
+  function lastTestInStack() {
+    return testStack.length ? testStack[testStack.length - 1] : null
   }
 
-  function resetDiagnosticMessage() {
-    diagnosticMessage = ''
+  function appendDiagnosticMessage(data) {
```

Comment:
I reordered these when I was going to use lastTestInStack.  Can put them back in the previous order if this is confusing.